jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ProjectAccountSettingsManager.kt

```diff
@@ -64,6 +64,12 @@ abstract class ProjectAccountSettingsManager(private val project: Project) : Sim
                         changeConnectionSettings(null, selectedPartition, selectedRegion)
                     }
                 }
+
+                override fun providerModified(identifier: ToolkitCredentialsIdentifier) {
+                    if (selectedCredentialIdentifier == identifier && connectionState is ConnectionState.InvalidConnection) {
```

Comment:
Should this always validate? If they modified the provider at all we should probably check again. Right now this fixes going from bad->good, but not good->bad

Reply:
I did think about it - my thought was do we want to disrupt an apparently stable connection? We would temporarily remove the explorer while we do the validate - and if they're off the network or something it'll prevent them from seeing the resources they could see before.

I can be convinced though - maybe it would be less confusing - and probably also makes sense in an MFA world.

Reply:
I (believe) we validate the profile actually has changes so editing Profile Foo will not trigger Bar to be modified.

If that is true, it would imply to me that they are making an active decision to mess with it.

(We may have a gap here though in source_profile chains, I am not sure we notify the consumers of the source_profile) has been modded if the source is altered)

Reply:
Will test the source profile